src/rest/HTTPError.js

```diff
@@ -0,0 +1,33 @@
+class HTTPError extends Error {
+  constructor(message, name, code, method, path) {
+    super(message);
+
+    /**
+     * The name of the error
+     * @type {string}
+     */
+    this.name = name;
+
+    /**
+     * HTTP error code returned from the request
+     * @type {number}
+     */
+    this.code = code || 500;
+
+    /**
+     * The HTTP method used for the request
+     * @type {string}
+     */
+    this.method = method;
+
+    /**
+     * The path of the request relative to the HTTP endpoint
+     * @type {string}
+     */
+    this.path = path;
+
+    if (Error.captureStackTrace) Error.captureStackTrace(this, this.constructor);
```

Comment:
If https://github.com/discordjs/discord.js/pull/2744 lands first, this line could be removed since it would capture the stack trace twice.